service/worker/scanner/scanner.go

```diff
@@ -289,3 +290,22 @@ func GetScannerContext(ctx context.Context) (scannerContext, error) {
 	}
 	return val, nil
 }
+
+// MaxTasksPerJob maximum tasks to process before allowing another task list to run
+
+func (cfg *Config) MaxTasksPerJob() int {
+	return cfg.MaxTasksPerJobFn()
+}
+
+// GetOrphanTasksPageSize the number of orphans to query in one call
+func (cfg *Config) GetOrphanTasksPageSize() int {
+	if cfg.GetOrphanTasksPageSizeFn != nil {
+		return cfg.GetOrphanTasksPageSizeFn()
+	}
+	return 16
```

Comment:
We don't set default this way. Can you check other configuration(like in other scanners). I think these tree methods can be removed if following the current convention.

Reply:
actually you have set default values in service.go. So you just need to remove the three methods.

Reply:
I put this there in deference to a test failure. For some reason during tests the default was not set causing a nilpointerexception. I couldn't work out where it was supposed to be set in that test - actually I couldn't even tell exactly which test was failing from the test output - so I gave up and did this.

The three methods must be there so that the config object implements the Config interface. This was the least intrusive way to introduce these dynamic config vars to scanner that I could find.

Reply:
Yeah I know what you are looking for. 
So you will create default like this: https://github.com/uber/cadence/blob/master/service/worker/scanner/history/scavenger_test.go#L73